app/controllers/teams_controller.rb

```diff
@@ -59,55 +55,75 @@ def destroy
 
   # Assigns team ownership to a specific team member
   def assign_ownership
-    unless owners_include_user
-      team_owners << user
-      redirect_to @team
-    end
+    assign_ownership_to_member
+    redirect_to @team
   end
 
   def delete_ownership
-    delete_owner_if_existing
+    delete_ownership_from_member
```

Comment:
Retrieve user who's ownership should be deleted as parameter and pass it as argument here:
`delete_ownership_from_member User.find(params[:team_member])`